src/test/java/com/google/cloud/teleport/templates/PubsubToSplunkTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2021 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.teleport.templates;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+/** Unit tests for {@link PubSubToSplunk}. */
+public class PubsubToSplunkTest {
+
+  private static final Gson GSON = new Gson();
+
+  /** Tests whether a Pub/Sub message with a {@link String} payload is parsed correctly. */
+  @Test
+  public void testStringPubsubMessage() {
+    String payload = "payload";
+    Map<String, String> attributes = ImmutableMap.of("key", "value");
+
+    PubsubMessage pubsubMessage = new PubsubMessage(payload.getBytes(), attributes);
+
+    String formattedPubsubMessage = PubSubToSplunk.formatPubsubMessage(pubsubMessage);
+
+    assertThat(formattedPubsubMessage).contains(PubSubToSplunk.DATA);
+    assertThat(formattedPubsubMessage).contains(PubSubToSplunk.ATTRIBUTES);
+
+    JsonObject messageJson = GSON.fromJson(formattedPubsubMessage, JsonObject.class);
+    assertThat(messageJson.get(PubSubToSplunk.DATA).getAsString()).isEqualTo(payload);
+  }
+
+  /** Tests whether a Pub/Sub message with JSON payload is parsed correctly. */
+  @Test
+  public void testJsonPubsubMessage() {
+    String payload =
+        "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"string\",\"ipsum\":\"id\"}}}";
```

Comment:
Not sure if this would make a difference, but consider adding a special character in one of the fields/keys like `@type` which is part of protoPayload documented here https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry

Reply:
Done